### DISCUSSION TITLE
Subset error in vignette

### DISCUSSION THREAD

**Discussion by GCAT01** [satijalab/seurat#9311] [Q&A]
I am following the "Analysis, visualization, and integration of spatial datasets with Seurat" vignette. I am at the section "Subset out anatomical regions" when I run the first subset command, I get the following:
```
> cortex <- subset(brain, idents = c(1, 2, 3, 4, 6, 7))
Warning messages:
1: Not validating Centroids objects 
2: Not validating Centroids objects 
3: Not validating FOV objects 
4: Not validating FOV objects 
5: Not validating FOV objects 
6: Not validating FOV objects 
7: Not validating FOV objects 
8: Not validating FOV objects 
9: Not validating Seurat objects 
```
Then, when running a further subset, it throws an error:
```
> cortex <- subset(cortex, anterior1_imagerow > 400 | anterior1_imagecol < 150, invert = TRUE)
Error in `FetchData()`:
! None of the requested variables were found: 
```
I have seen a few other people having issues with subset although I can't seem to find a solution that works for me in this situation.

**Comment by suanzaoren** [comment]
Hi , I have the same absolutely error!  I also hope someone could give me a help. If you have resolved this issue, please tell me, thanks.

**Comment by suanzaoren** [comment]
> I am following the "Analysis, visualization, and integration of spatial datasets with Seurat" vignette. I am at the section "Subset out anatomical regions" when I run the first subset command, I get the following:
> 
> ```
> > cortex <- subset(brain, idents = c(1, 2, 3, 4, 6, 7))
> Warning messages:
> 1: Not validating Centroids objects 
> 2: Not validating Centroids objects 
> 3: Not validating FOV objects 
> 4: Not validating FOV objects 
> 5: Not validating FOV objects 
> 6: Not validating FOV objects 
> 7: Not validating FOV objects 
> 8: Not validating FOV objects 
> 9: Not validating Seurat objects 
> ```
> 
> Then, when running a further subset, it throws an error:
> 
> ```
> > cortex <- subset(cortex, anterior1_imagerow > 400 | anterior1_imagecol < 150, invert = TRUE)
> Error in `FetchData()`:
> ! None of the requested variables were found: 
> ```
> 
> I have seen a few other people having issues with subset although I can't seem to find a solution that works for me in this situation.

Hi,  Now I have a possible solution:
# Extract coordinates (assuming the image name is "anterior1")
spatial_coords <- GetTissueCoordinates(cortex, image = "anterior1")

Rename a column (optional: Make sure the column name matches the subset condition)
colnames(spatial_coords) <- c("anterior1_imagerow", "anterior1_imagecol")

# Adds coordinates to metadata
cortex <- AddMetaData(cortex, metadata = spatial_coords[,-3])
# Check the column names of the metadata to make sure they are correct
colnames(cortex@meta.data)
